Description: Work around broken compilers (like fpc) on armhf
 patches/arm/local-vfp-sysdeps.diff: Force the Tag_ABI_HardFP_use
 and Tag_ABI_VFP_args EABI tags when building for armhf, so compilers
 only linking crti.o will generate what appear to be armhf binaries.
Author: Adam Conrad <adconrad@debian.org>
Forwarded: no

--- eglibc-2.16.orig/ports/sysdeps/arm/sysdep.h
+++ eglibc-2.16/ports/sysdeps/arm/sysdep.h
@@ -117,6 +117,13 @@
    the caller.  */
 	.eabi_attribute 24, 1
 
+#ifdef __ARM_PCS_VFP
+/* Tag_ABI_HardFP_use: This code uses hard floating point */
+	.eabi_attribute 27, 3
+/* Tag_ABI_VFP_args: This code stores FP arguments in VFP registers */
+	.eabi_attribute 28, 1
+#endif  /* __ARM_PCS_VFP */
+
 /* The thumb2 encoding is reasonably complete.  Unless suppressed, use it.  */
 	.syntax unified
 # if defined(__thumb2__) && !defined(NO_THUMB)
